feat(profiling): add support for processing profile chunks with vroomrs

viglia opened the pull request:
This PR adds the support for processing profile chunks with the python extension module `vroomrs` instead of sending it over to the `vroom` service.


Depends on https://github.com/getsentry/getsentry/pull/17308

Commit:
add profiles processing with vroomrs

add the capability (behind a feature-flag) to process the profile chunks
directly in sentry, through the vroomrs, PyO3 module, instead of sending
them over to the vroom service.

Commit:
kafka messages fixes

Commit:
fix vroomrs import untyped mypy warning

getsantry[bot] committed:
:snowflake: re-freeze requirements

Commit:
fix feature flag name

Commit:
Merge remote-tracking branch 'origin' into viglia/feat/add-vroomrs-chunk-processing

asottile-sentry commented on src/sentry/profiles/task.py:
```diff
@@ -9,7 +10,9 @@
 import msgpack
 import sentry_sdk
-from arroyo.backends.kafka import KafkaProducer, build_kafka_configuration
+import vroomrs  # type: ignore[import-untyped]
```
new things should not be untyped.  please fix this by adding stubs in vroomrs

Commit:
bump vroomrs to version 0.1.4 which contains the python stubs

this allow us to remove `type: ignore[import-untyped]` and let mypy do
its type checking.
Also, code auto-complete will be available in the IDE.

getsantry[bot] committed:
:snowflake: re-freeze requirements

Zylphrex commented on requirements-dev-frozen.txt:
Any idea why all these unrelated packages got bumped too? Ideally we dont touch them in this PR

viglia replied:
I didn't bump them myself. I think it was an unfortunate mix of not having the lates master branch merged and the `getsentry` _Github_ app doing its own things trying to update the _requirements-frozen_ list.

I've re-merged the latest master and it seems to have fixed that.

Zylphrex commented on src/sentry/profiles/task.py:
```diff
@@ -1189,3 +1200,100 @@ def track_latest_sdk(project: Project, profile: Profile) -> None:
+                )
+                profile_functions_producer.produce(topic, payload)
+            return True
+        except Exception as e:
```
This has no retries built-in as far as I can tell. Is this something we should look into?

viglia replied:
I'm quite surprised this was not added to the `SingletoneProducer` options in Arroyo.

I've followed up with the team maintaining Arroyo [here](https://sentry.slack.com/archives/C04E4723EJ1/p1746451747689219), and I proposed to add the retries, Backoff.ms, maxBackoff.ms and delivery.timeout.ms to the list of allowed options. 

Let's see what they say.

viglia replied:
So, long story short: it has built-in retries for the simple fact that it's a wrapper around rdkafka, which already has default value for retries, minBackoff, maxBackoff, delivery.timeout etc. etc. see here for the default value: https://docs.confluent.io/platform/current/clients/librdkafka/html/md_CONFIGURATION.html

With that said I still think we should add them as configurable option in `SUPPORTED_KAFKA_CONFIGURATION` so that if we or anyone else want to specialize those options for their own producer they can, but I would do this in a separate PR has it has nothing to do with this PR and, as for now, I think we're safe to rely on the defaults as for all the other running producers in Sentry.

Zylphrex replied:
Im more curious about retries on GCS

viglia replied:
Yes, the GCS filestore implementation does support retries.

Zylphrex commented on src/sentry/profiles/task.py:
```diff
@@ -204,8 +208,15 @@ def process_profile_task(
-        return
+    if (
+        features.has("projects:continuous-profiling-vroomrs-processing", project)
+        and "profiler_id" in profile
```
Just checking but does vroomrs handle android chunks already?

viglia replied:
Correct, vroomrs fully supports profile chunks (Android and Sample format).

Commit:
Merge remote-tracking branch 'origin' into viglia/feat/add-vroomrs-chunk-processing

Commit:
Merge remote-tracking branch 'origin' into viglia/feat/add-vroomrs-chunk-processing

Commit:
Merge remote-tracking branch 'origin' into viglia/feat/add-vroomrs-chunk-processing

Commit:
Merge remote-tracking branch 'origin' into viglia/feat/add-vroomrs-chunk-processing
